app/ui-react/syndesis/src/modules/connections/components/ConnectionsWithToolbarNew.tsx

```diff
@@ -0,0 +1,126 @@
+import { Connection } from '@syndesis/models';
+import {
+  ConnectionsListView,
+  IActiveFilter,
+  IConnectionsListViewProps,
+  IFilterType,
+  ISortType,
+} from '@syndesis/ui';
+import { WithListViewToolbarHelpers } from '@syndesis/utils';
+import * as React from 'react';
+import { Translation } from 'react-i18next';
+import i18n from '../../../i18n';
+import resolvers from '../resolvers';
+import { Connections, IConnectionsProps } from './Connections';
+
+function getFilteredAndSortedConnections(
+  connections: Connection[],
```

Comment:
we are not dealing with `Connection` objects here, but with `IUIIntegrationStep` which is a "meta object" that can hold `Connection`s, `StepKind`s and `Extension`s